File: Examples/VisualBasic/FormattingAndManipulatingOCR/RecognizeWhiteListedCharacters.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.OCR

Namespace Aspose.OCR.Examples.VisualBasic.FormattingAndManipulatingOCR
    Public Class RecognizeWhiteListedCharacters
        Public Shared Sub Run()
            ' ExStart:RecognizeWhiteListedCharacters
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_OCR()

            'Initialize an instance of OcrEngine
            Dim ocrEngine As New OcrEngine()

            'Retrieve the OcrConfig of the OcrEngine object
            Dim ocrConfig As OCRConfig = ocrEngine.Config

            'Set the Whitelist property to recognize numbers only
            ocrConfig.Whitelist = New Char() {"1"c, "2"c, "3"c, "4"c, "5"c, "6"c, "7"c, "8"c, "9"c, "0"c}

            'Set the Image property of OcrEngine object
            ocrEngine.Image = ImageStream.FromFile(dataDir & "Sampleocr.bmp")

            'Call the Process method to retrieve the results
            ocrEngine.Process()
            ' ExEnd:RecognizeWhiteListedCharacters
        End Sub
    End Class
End Namespace

File: Examples/CSharp/FormattingAndManipulatingOCR/RecognizeWhiteListedCharacters.cs
using System;
using System.IO;
using Aspose.OCR;

namespace Aspose.OCR.Examples.CSharp.FormattingAndManipulatingOCR
{
    public class RecognizeWhiteListedCharacters
    {
        public static void Run()
        {
            // ExStart:RecognizeWhiteListedCharacters
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_OCR();

            // Initialize an instance of OcrEngine
            OcrEngine ocrEngine = new OcrEngine();

            // Retrieve the OcrConfig of the OcrEngine object
            OCRConfig ocrConfig = ocrEngine.Config;

            // Set the Whitelist property to recognize numbers only
            ocrConfig.Whitelist = new char[] { '1', '2', '3', '4', '5', '6', '7', '8', '9', '0' };

            // Set the Image property of OcrEngine object
            ocrEngine.Image = ImageStream.FromFile(dataDir+ "SampleNumbers.jpg");

            // Call the Process method to retrieve the results
            ocrEngine.Process();

            Console.WriteLine("Output: " + ocrEngine.Text);
            // ExEnd:RecognizeWhiteListedCharacters
        }
    }
}